src/main/java/org/dwcj/controls/AbstractDwcControl.java

```diff
@@ -104,30 +127,61 @@ public String getAttribute(String attribute) {
         return attributes.get(attribute);
     }
 
+    
+    protected AbstractDwcControl setReadOnly(boolean readOnly){
+        if(ctrl != null){
+            try{
+                ctrl.setEnabled(readOnly);
+            } catch(BBjException e){
+                e.printStackTrace();
+            }
+            this.readOnly = readOnly;
+        }
+        return this;
+    }
+
+    protected Boolean isReadOnly(){
+        if(ctrl != null){
+            return this.readOnly;
+        }
+        return null;
```

Comment:
Agreed on both of these - on my current branch I've made the latter change, and agree we should discuss what the defaults for some of these properties are and return those instead of null by default.